File: KOTLIN/zadanie_3.kt
fun main() {
    val input = readLine()!!.split(" ")
    val output = input.mapIndexedNotNull { index, number ->
        val digits = number.map { it.toString().toInt() }
        val sum = digits.sum()
        val product = digits.reduce { acc, d -> acc * d }
        if (sum < product) index else null
    }
    println(output.joinToString(" "))
}

File: SWIFT/zadanie_3.swift
import Foundation

if let input = readLine() {
    let numbers = input.split(separator: " ")
    var output: [Int] = []
    for (index, number) in numbers.enumerated() {
        let digits = number.compactMap { Int(String($0)) }
        let sum = digits.reduce(0, +)
        let product = digits.reduce(1, *)
        if sum < product {
            output.append(index)
        }
    }
    print(output.map(String.init).joined(separator: " "))
}
